### ISSUE TITLE
[Feature Request] Adjust lines to scroll with mouse wheel

### ISSUE THREAD

**Issue by AnsonH** [akiyosi/goneovim#223]
I found the speed of scrolling too slow in GoNeovim. It seems like the default number of lines to scroll with mouse wheel is set to 1, because scrolling down is mapped to the keystroke `1<C-e>`, which is equivalent to scrolling 1 line down.

It would be cool if there is a new option in `settings.toml` where I can customize the scrolling speed. For example:
```toml
## (New option) Lines to scroll with mouse wheel
LinesToScroll = 5
```
Then, scrolling down would be mapped to the keystroke `5<C-e>` such that it scrolls down 5 lines instead of 1 line.
Thank you very much!

**Comment by akiyosi** [comment]
Hi, Thanks for this request!
I have tried to implement this feature.
Could you please get the latest binary from the following and check if it works?
https://github.com/akiyosi/goneovim/actions/runs/866089915

**Comment by AnsonH** [comment]
Sorry I don't have Qt and Go installed in my computer as I'm using the pre-built executable file on Windows 10.

It would be appreciated if anyone who already have his/her development environment properly set up can help check that it works

**Comment by akiyosi** [comment]
@AnsonH 
Hey, You can get the latest binaries built from Github Actions :)
<img width="1138" alt="2021-05-22" src="https://user-images.githubusercontent.com/8478977/119222787-04028e00-bb31-11eb-92bc-770962c48941.png">

**Comment by AnsonH** [comment]
Wow thanks for letting me know! 
I've added `LineToScroll = 5` in my `settings.toml` and it is successfully mapped to `5<C-e>` and `5<C-y>` while scrolling. However, may I ask is scrolling down wrongly mapped to `5<C-y>` and it should be `5<C-e>`  instead? That's because `5<C-e>` moves the screen 5 lines upwards, which is similar to scroll down for 5 lines. Similarly, should scrolling up be mapped to `5<C-y>` instead?

**Comment by akiyosi** [comment]
Thanks for the feedback!

This problem is generally assumed in situations where the mouse wheel is used in Windows or the touch pad is used in MacOS, and I could not find a uniform solution for these situations.
So I have added an option `ReversingScrollDirection` in `[Editor]` section. Setting this option to `true` reverses scrolling with the mouse wheel.
I would appreciate it if you could confirm the behavior.

https://github.com/akiyosi/goneovim/actions/runs/866928302

Please note that when `SmoothScroll` is `true`, fast mouse wheel scrolling will result in skipped buffer rendering. I'm looking for ways to improve this, but there is no effective solution at this time. If this bothers you, please disable `SmoothScroll`.

**Comment by AnsonH** [comment]
I tested out `ReversingScrollDirection` and it's working great! Thanks a lot!

Just a friendly reminder, don't forget to update the [Usage page of the wiki] (https://github.com/akiyosi/goneovim/wiki/Usage#configuration) when this feature is released :)

**Comment by akiyosi** [comment]
Thanks for confirming!

> Just a friendly reminder, don't forget to update the Usage page of the wiki when this feature is released :)

Of course, I will update the wiki in the next release. Thanks!

**Comment by akiyosi** [comment]
This issue was closed with the release of v0.4.12.
